### PR TITLE
Add fuzz testing setup

### PR DESCRIPTION
**Pull request by papandreou**
This is how I found out what to fix in https://github.com/devongovett/exif-reader/pull/5

This PR will be marked as failing, but when combined with the fixes in https://github.com/devongovett/exif-reader/pull/5 it will succeed.

### CODE REVIEW

**Comment by lovell** [comment]
@papandreou Thank you for these tests. I tried running them locally after merging #5 but see a few failures that all look a bit like:

```sh
  expected Buffer([0x45, 0x78, 0x69, 0x66, 0x00, 0x00, 0x4D, 0x4D, 0x00, 0x2A, 0x00, 0x00, 0x00, 0x08, 0x00, 0x0B /* 8142 more */ ]) to either parse or throw documented error
    Threw unexpected error: TypeError [ERR_INVALID_ARG_TYPE]: The "offset" argument must be of type number. Received type string
        at validateNumber (internal/validators.js:130:11)
```

I realise this PR was opened a while ago, so no worries if you're unable to do so now, but if you do have the time please can you see if there's anything obvious that can be updated/fixed so we can merge this useful feature.

**Comment by papandreou** [comment]
@lovell, got it back into shape now and found another bug (#14) :hugs:

File: index.js [hunk 1]
```diff
@@ -187,20 +191,19 @@ function readValue(buffer, offset, bigEndian, type) {
...
-  var date = new Date(1970, 0, 2); // to prevent unexpected month change after setUTCMonth()
-  date.setUTCFullYear(match[1]);
-  date.setUTCMonth(match[2] - 1);
-  date.setUTCDate(match[3]);
-  date.setUTCHours(match[4]);
-  date.setUTCMinutes(match[5]);
-  date.setUTCSeconds(match[6]);
-  date.setUTCMilliseconds(0);
-  return date;
+  return new Date(Date.UTC(
```
Comment (lovell) [inline index.js]: Was this change intentional? It looks like it reverts some of commits https://github.com/devongovett/exif-reader/commit/ea5ed3d307bae40578d820cc5e27d5fcc1ce2c9c and https://github.com/devongovett/exif-reader/commit/be0016f4e4d8dda55c7acaba6276e1f3ccd0c24a

**Review by lovell — COMMENTED** [review]
Thank you for updating this, there's one comment inline but otherwise this is good to go.

File: index.js (hunk 1 above)
Comment (papandreou) [reply index.js]: Yeah, it was intentional. It fixes the same issue as be0016f4e4d8dda55c7acaba6276e1f3ccd0c24a (that UTC timestamps were interpreted as local time because of `new Date`). If I recall correctly, I made the change to get the test suite to even pass on my computer, which must have been in GMT+2 back on July 16 2016 when I made the change :sweat_smile:

I think converting to UTC in one go is slightly nicer, but either fix does the job. I've undone all the changes to index.js here and made a separate PR now: #15

**Comment by lovell** [comment]
Brilliant, thank you Andreas.
